File: backend/models/vehicle.py
from enum import Enum


class VehicleTypes(Enum):
    COMPACT = ("compact", 30, 150)
    MEDIUM = ("medium", 30, 150)
    FULL_SIZE = ("full-size", 30, 150)
    CLASS_1_TRUCK = ("class 1 truck", 60, 250)
    CLASS_2_TRUCK = ("class 2 truck", 120, 700)

    def __init__(self, vehicle_type, servicing_time, charge):
        self.type = vehicle_type
        self.servicing_time = servicing_time
        self.charge = charge

    def __str__(self):
        return self.type


def get_vehicle_by_type(vehicle_type: str) -> "VehicleTypes":
    for vehicle in VehicleTypes:
        if vehicle.type == vehicle_type:
            return vehicle
    return None

File: frontend/src/models/vehicle.ts
export interface Vehicle {
    vehicle_type: string;
    servicing_time: number;
    charge: number;
}

export enum VehicleType {
    COMPACT = "Compact",
    MEDIUM = "Medium",
    FULL_SIZE = "Full-size",
    CLASS_1_TRUCK = "Class 1 truck",
    CLASS_2_TRUCK = "Class 2 truck",
}

export const VehicleDetails: { [key in VehicleType]: { servicing_time: number, charge: number } } = {
    [VehicleType.COMPACT]: { servicing_time: 30, charge: 150 },
    [VehicleType.MEDIUM]: { servicing_time: 30, charge: 150 },
    [VehicleType.FULL_SIZE]: { servicing_time: 30, charge: 150 },
    [VehicleType.CLASS_1_TRUCK]: { servicing_time: 60, charge: 250 },
    [VehicleType.CLASS_2_TRUCK]: { servicing_time: 120, charge: 700 },
};